core/src/main/java/org/apache/iceberg/util/ContentFileUtil.java

```diff
@@ -49,7 +50,21 @@ public static <F extends ContentFile<K>, K> K copy(
     }
   }
 
+  /**
+   * @deprecated since 1.7.0, will be removed in 2.0.0; use {@link
```

Comment:
I think it's fine to remove this in 1.8.0 instead of 2.0.0 since this is iceberg-core?

Reply:
Agree, we can drop it in 1.8.

I wouldn't also mind not deprecating this for now to see if there would be any use cases.

Reply:
I thought about this a bit more,  I originally deprecated it because I did think there's no need to have both methods. However, after more thought I think it makes sense to not deprecate for now since it isn't super obvious (at least to me) we will never need this.

 If at a certain point we realize it really isn't needed we can easily deprecate at that point (especially since it's in core like @nastra mentioned). However, if we deprecate and realize we need it in some cases then we're in an awkward state.